documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,258 @@ Control characters are not allowed in a specification.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2015" />R2015 ParameterNotDefinedInGlobalParameters
+**Output Message**: Parameter "{0}" is referenced but not defined in the global parameters section of Service Definition
+
+**Description**: Per ARM guidelines, if `subscriptionId` is used anywhere as a path parameter, it must always be defined as global parameter. `api-version` is almost always an input parameter in any ARM spec and must also be defined as a global parameter.
+
+**Why the rule is important**: To reduce duplication, maintain consistent structure in ARM specifications, and ensure `subscriptionId` and `api-version` are created as client properties in the generated code.
+
+**How to fix the violation**: Ensure `subscriptionId` and `api-version` are declared in the global parameters section of the document.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2020" />R2020 RequiredPropertiesMissingInResourceModel
+**Output Message**: Model definition '{0}' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.
+
+**Description**: Per ARM guidelines, a `Resource` model must have the `name`, `id` and `type` properties defined as `readOnly` in its hierarchy.
+
+**Why the rule is important**: `name`, `type` and `id` are readonly properties set on the service end. Also, per ARM guidelines each `Resource` type model must have these properties defined in its hierarchy. An example `Resource` definition can be found [here](https://github.com/Azure/azure-rest-api-specs-pr/blob/master/common-types/resource-management/v1/types.json#L3).
+
+**How to fix the violation**: Ensure the `Resource` type model has the properties `name`, `type` and `id` and they are marked as `readOnly:true`. 
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2056" />R2056 RequiredReadOnlyProperties
+**Output Message**: Property '{0}' is a required property. It should not be marked as 'readonly'.
+
+**Description**: A model property cannot be both `readOnly` and `required`. A `readOnly` property is something that the server sets when returning the model object while required is a property required to be set when sending it as a part of the request body.
```

Comment:
you may want to add quotes around "required" in its second instance in the sentence too.